### ISSUE TITLE
Add a project logo and favicon to build brand identity

### ISSUE THREAD

**Issue by sachncs** [sachncs/regulo#20]
## Problem

The repository has no logo, no favicon, no `docs/img/` directory, no diagrams, and no SVG. The README contains only inline `img.shields.io` badges. Compared to the brand surface of a typical OSS Python library (e.g. `numpy`, `scikit-learn`, `pytorch`), `regulo` reads as a plain-text research artefact rather than a product.

## Evidence

- `grep -rn '<img' --exclude-dir=.git .` → no `<img>` tags except in badges.
- `ls docs/img 2>&1` → No such file or directory
- `ls logo* favicon* 2>&1` → No such file or directory
- `.gitignore:91-92` lists `*.png`, `*.jpg`, `*.svg` as binary/text, but no such files exist in the tree.
- `pyproject.toml` has no `[project.urls].Logo` entry.

## Root Cause

The project was set up for a research paper reproduction; visual identity was not part of the original scope.

## Impact

- Brand recall is low — visitors cannot anchor the project to a memorable visual.
- Documentation pages (when GitHub Pages is enabled) will inherit the mkdocs-material default theme with no project customisation.
- Marketing / outreach (HN, Twitter, conference talks) has no logo to drop into slides or screenshots.

## Recommendation

Add at least one visual asset to the repository:

1. **A monochrome wordmark or symbol SVG** (`docs/img/logo.svg` or `docs/img/logo.png`). Suggested concept: a small diagonal scatter of dots representing correlated features with a regularising "halo" line — evocative of Covridge shrinkage.
2. **A favicon** (`docs/img/favicon.ico` or `docs/img/favicon.svg`).
3. **Reference the logo from `README.md`** (line ~10, before or after the title):
   ```markdown
   ![regulo](docs/img/logo.svg)
   ```
4. **Reference the logo from `pyproject.toml`**:
   ```toml
   [project.urls]
   Logo = "https://raw.githubusercontent.com/sachncs/regulo/master/docs/img/logo.svg"
   ```

This is a low-priority brand improvement; pair with the GitHub Pages enablement issue so the logo actually shows up on the docs site.

## Acceptance Criteria

- [ ] `docs/img/logo.svg` (or `.png`) is committed.
- [ ] `README.md` references the logo.
- [ ] Logo is monochrome and legible at 32×32 (favicon size).
- [ ] Visual style matches the technical / academic tone of the README (no decorative gradients).

## Verification

- [ ] Manual: render the README; confirm the logo appears.
- [ ] `git log --follow docs/img/logo.svg` records the introduction.

## Risk

None — asset addition only.

**Comment by sachncs** [comment]
Resolved in PR #29.
